huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go

```diff
@@ -42,385 +55,1017 @@ func ResourceDwsCluster() *schema.Resource {
 				Computed: true,
 				ForceNew: true,
 			},
-
 			"name": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
-			},
-
-			"network_id": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The cluster name.`,
 			},
-
 			"node_type": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The flavor of the cluster.`,
 			},
-
 			"number_of_node": {
-				Type:     schema.TypeInt,
-				Required: true,
-			},
-
-			"security_group_id": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: `Number of nodes in a cluster.`,
 			},
-
 			"user_name": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Administrator username for logging in to a data warehouse cluster.`,
 			},
-
 			"user_pwd": {
-				Type:      schema.TypeString,
-				Required:  true,
-				Sensitive: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Administrator password for logging in to a data warehouse cluster.`,
 			},
-
 			"vpc_id": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The VPC ID.`,
+			},
+			"network_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The subnet ID.`,
+			},
+			"security_group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The security group ID.`,
 			},
-
 			"availability_zone": {
-				Type:     schema.TypeString,
-				Computed: true,
-				Optional: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The availability zone in which to create the cluster instance. `,
+			},
+			"version": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				Description:  "schema: Required",
+				RequiredWith: []string{"number_of_cn", "volume"},
+			},
+			"number_of_cn": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "schema: Required",
 			},
-
 			"port": {
-				Type:     schema.TypeInt,
-				Computed: true,
-				Optional: true,
-				ForceNew: true,
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     8000,
+				Description: "schema: Required",
```

Comment:
If the param has a default value, I think we can keep it as optional.

Reply:
ok